fairscale/optim/oss.py

```diff
@@ -613,5 +615,9 @@ def _setup_bucket_strategy(self) -> None:
                 self.buckets[device][dst_rank].global_rank = self.global_rank
 
         # Determine the max work handles in flight:
-        # - all the direct reduce/broadcast + 1 bucket
-        self._max_work_handles = sum(not value for value in self.should_bucket_param.values()) + 1
+        # - all the direct reduce/broadcast
+        self._max_work_handles = sum(not value for value in self.should_bucket_param.values())
```

Comment:
this was wrong:
- if no bucketing, then the +1 was wrong (no more requests, just the direct ones)
- if bucketing, the +1 was mostly wrong (one more request per bucket, which means one per device per rank)